js/sales.js

```
@@ -2,9 +2,59 @@
 
 console.log('sales.js loaded');
 
-buildHeader();
-var rowPosition = document.getElementById('cookie_sale_body'); //where do we want to put the data rows
-buildDataRows(rowPosition);
+window.addEventListener('load', builtTableOnLoad);
+
+var newCookieStore = {};
+var newLocation = document.getElementById('newLocation');
+var newAddress = document.getElementById('newAddress');
+var newOpenHour = document.getElementById('newOpenHour');
+var newCloseHour = document.getElementById('newCloseHour');
+var newMinHourlyCustomers = document.getElementById('newMinHourlyCustomers');
+var newMaxHourlyCustomers = document.getElementById('newMaxHourlyCustomers');
+var newAvgCookiesPerSale = document.getElementById('newAvgCookiesPerSale');
+var createLocationButton = document.getElementById('createLocationButton');
+
+createLocationButton.addEventListener('click', function(event){
+  if(breakOnEmpty()){
+    alert('Please fill in the Location and Address fields.');
+    return;
+  }
+  buildCookieStore();
+  buildRowForNewCookieStore();
+  event.preventDefault();
```

I mentioned this as an afterthought in class, but you should put this as high up in the function as you know you want to `preventDefault()`. In this case, that would be at the very top, otherwise you won't `preventDefault()` if there's an error (though the `alert` hides this).